Feature/9 votos panel

NoSoyDani opened the pull request:
Añade al panel de control la funcionalidad necesaria para mostrar los resultados obtenidos de una votación en la sección "Votos", introduciendo la id de la votación en la URL.

NoSoyDani committed:
feat: Añade funcionalidad base

Se han creado los archivos necesarios para realizar la funcionalidad
así como, se ha creado la llamada a la API necesaria para mostrar datos
sobre las votaciones.

NoSoyDani committed:
feat: Añade funcionalidad deseada

Se ha implementeado la funcionalidad necesaria para que introduciendo
el id de la votación en la url del panel de control, se muestren en la
pestaña correspondiente a "Votos" los resultados finales de la votación.

Se ha implementado información adicional sobre la votación en la pestaña
"Votos"

NoSoyDani committed:
fix: Eliminada dependencia de archivos innecesaria

Se ha eliminado la dependencia al archivo Voting.js que causaba error.

manmunesp12 reviewed (approved):
Funcionalidad comprobada, panel de voto implementado correctamente.

Javitoox reviewed (changes requested):
Cambios requeridos:
- Obtención de datos en tiempo real.
- La obtención de los votos se debe hacer sin modificar la url.
- Añadir información adicional a los votos.

NoSoyDani committed:
feat: Añade funcionalidad de refresco en tiempo real 

 Se ha añadido la funcionalidad necesaria para visualizar los resultados de las votaciones en tiempo real.

NoSoyDani committed:
feat: Añadida nueva consulta en la API

Se ha añadido una nueva funcionalidad a la API que permite
visualizar todas las votaciones, así como, datos adicionales
de las mismas.

NoSoyDani committed:
Merge branch 'feature/9-votos_panel' of github.com:Full-Tortuga/decide-full-tortuga-almacenamiento into feature/9-votos_panel

NoSoyDani committed:
feat: Añade funcionalidad listado de votaciones

Se ha explotado la consulta de la API que devuelve
el conjunto de votaciones, para mostrar en el
panel de control el resultado de todas las votaciones
en tiempo real.

NoSoyDani committed:
fix: Elimina deslizable sección votos

Se ha eliminado la funcionalidad del deslizable
que permitia seleccionar la votación, ya que este
no aporta una funcionalidad relevante.

NoSoyDani commented:
Se ha implementado lo solicitado en la última request change.

Javitoox reviewed (dismissed):
Cambios requeridos que quedan por hacer:

Añadir información adicional a los votos.

Además al revisar la información mostrada, la funcionalidad requería mostrar los votos. Actualmente solo se puede ver información genérica de la votación y no se puede ver información acerca de los votos. Para ello se debería de modificar el backend.

Javitoox commented:
La funcionalidad extra requerida es referente al censo, ya que actualidad la vista de votos muestra la misma información que el propio panel de administración de Django. Para la vista de votaciones se implementó en su momento información extra relacionada con el censo tal y como requería el incremento, en caso contrario el incremento no sería notable. Por lo tanto, teniendo en cuenta que la vista de votas es una copia de la vista de votaciones y no añade ninguna información nueva, se requiere mostrar algún tipo de información extra relacionada con el censo, ya que esto es fácilmente modificable en el backend de decide y no pone en entredicho la privacidad del usuario, ya que es información relacionada con el censo y dicho censo con tiene el id de la votación a la que pertenece y el id del usuario.

dancasnar1 reviewed (approved):
Se ha comprobado el correcto funcionamiento de la funcionalidad.

Javitoox reviewed (dismissed):
Se requiere añadir lo comentado anteriormente. Actualmente no hay ningún incremento de la funcionalidad al ser una copia de la vista de votaciones.

NoSoyDani committed:
Merge branch 'main' into feature/9-votos_panel

Javitoox commented:
No hay ninguna funcionalidad añadida y se ha aceptado la pull request quitando mi revisión sin previo aviso. Tal y como bien se describe aquí: 
![image](https://user-images.githubusercontent.com/55922195/148211187-446855ee-1f94-451a-b4da-d9709b5499fc.png)
El incremento es añadir nueva información, ya que los otros incrementos de tiempo real y otras funcionalidades ya los había implementado anteriormente y solo ha sido replicar lo mismo en esta tarea. La descripción de la tarea debía haber sido redactada siguiendo la imagen anterior, ya que es la descripción propuesta por la asignatura.

En la API no hay nada modificado y tal y como se puede ver en la vista de votos se muestra exactamente lo mismo que en la vista de votaciones previamente implementada, por lo que esta vista ha sido un copia y pega de la vista de votaciones y sin añadir nada nuevo.

No se requería modificar ningún dato privado de los votos, solo datos del censo tal y como se muestra en la descripción anterior y esto se hacía fácilmente modificando un método en el backend, al igual que ha sido modificado para la vista de votaciones. Esta tarea llevaba desde el primer día que se empezó el proyecto y hubiera dado tiempo de sobra.

Además, ha sido esta pull request ha sido aceptada sin el consentimiento de la mayoría del grupo.
